indico_toolkit/indico_wrapper/indico_wrapper.py

```diff
@@ -57,13 +61,46 @@ def train_model(
             )
         )
 
-    def get_storage_object(self, storage_url):
+    def _retry(exceptions, wait=0.5):
+        def retry_decorator(fn):
+            @wraps(fn)
+            def retry_func(*args, **kwargs):
+                num_retries = 5
+                if "num_retries" in kwargs:
+                    num_retries = kwargs.get("num_retries")
+                retries = 0
+                while True:
+                    try:
+                        return fn(*args, **kwargs)
+                    except exceptions as e:
+                        if retries >= num_retries:
+                            raise e
+                        else:
+                            retries += 1
+                            print(
+                                f"Retrying function {fn.__name__} after exception {e}"
+                            )
+                            time.sleep(wait)
+                            continue
```

Comment:
why do you need continue here?

Reply:
I think we can also ditch the "retry" print statement- if we are building this in, unnecessary to call it out I'd say